--- /etc/nginx/conf.d/st2.template	2018-07-26 16:55:35.000000000 +0000
+++ /etc/nginx/conf.d/st2.conf	2018-07-26 17:06:30.200000000 +0000
@@ -18,8 +18,8 @@
 
   index  index.html;
 
-  access_log /var/log/nginx/st2webui.access.log combined;
-  error_log /var/log/nginx/st2webui.error.log;
+  access_log /proc/self/fd/1 combined;
+  error_log stderr;
 }
 
 server {
@@ -37,8 +37,8 @@
 
   index  index.html;
 
-  access_log            /var/log/nginx/ssl-st2webui.access.log combined;
-  error_log             /var/log/nginx/ssl-st2webui.error.log;
+  access_log            /proc/self/fd/1 combined;
+  error_log             stderr;
 
   add_header              Front-End-Https on;
   add_header              X-Content-Type-Options nosniff;
@@ -53,7 +53,7 @@
 
     rewrite ^/api/(.*)  /$1 break;
 
-    proxy_pass            http://127.0.0.1:9101/;
+    proxy_pass            ${ST2_API_URL};
     proxy_read_timeout    90;
     proxy_connect_timeout 90;
     proxy_redirect        off;
@@ -83,7 +83,7 @@
 
     rewrite ^/stream/(.*)  /$1 break;
 
-    proxy_pass  http://127.0.0.1:9102/;
+    proxy_pass  ${ST2_STREAM_URL};
     proxy_set_header Host $host;
     proxy_set_header X-Real-IP $remote_addr;
     proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
@@ -111,7 +111,7 @@
 
     rewrite ^/auth/(.*)  /$1 break;
 
-    proxy_pass            http://127.0.0.1:9100/;
+    proxy_pass            ${ST2_AUTH_URL};
     proxy_read_timeout    90;
     proxy_connect_timeout 90;
     proxy_redirect        off;
