--- /etc/nginx/conf.d/st2.template	2020-05-10 02:12:53.977869285 +0200
+++ /etc/nginx/conf.d/st2.conf	2020-05-10 02:13:07.373975299 +0200
@@ -1,6 +1,6 @@
 #
-# nginx configuration to expose st2 webui, redirect HTTP->HTTPS,
-# provide SSL termination, and reverse-proxy st2api and st2auth API endpoint.
+# nginx configuration to expose st2 webui, and reverse-proxy the st2api and
+# st2auth API endpoints.
 # To enable:
 #    cp ${LOCATION}/st2.conf /etc/nginx/sites-available
 #    ln -l /etc/nginx/sites-available/st2.conf /etc/nginx/sites-enabled/st2.conf
@@ -9,37 +9,14 @@
 server {
   listen *:80 default_server;
 
-  add_header Front-End-Https on;
   add_header X-Content-Type-Options nosniff;
 
-  if ($ssl_protocol = "") {
-       return 308 https://$host$request_uri;
-  }
-
-  index  index.html;
-
-  access_log /var/log/nginx/st2webui.access.log combined;
-  error_log /var/log/nginx/st2webui.error.log;
-}
-
-server {
-  listen       *:443 ssl;
-
-  ssl_certificate           /etc/ssl/st2/st2.crt;
-  ssl_certificate_key       /etc/ssl/st2/st2.key;
-  ssl_session_cache         shared:SSL:10m;
-  ssl_session_timeout       5m;
-  ssl_protocols             TLSv1 TLSv1.1 TLSv1.2;
-  ssl_ciphers               EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH:ECDHE-RSA-AES128-GCM-SHA384:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA128:DHE-RSA-AES128-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES128-GCM-SHA128:ECDHE-RSA-AES128-SHA384:ECDHE-RSA-AES128-SHA128:ECDHE-RSA-AES128-SHA:ECDHE-RSA-AES128-SHA:DHE-RSA-AES128-SHA128:DHE-RSA-AES128-SHA128:DHE-RSA-AES128-SHA:DHE-RSA-AES128-SHA:AES128-GCM-SHA384:AES128-GCM-SHA128:AES128-SHA128:AES128-SHA128:AES128-SHA:AES128-SHA:HIGH:!aNULL:!eNULL:!EXPORT:!DES:!MD5:!PSK:!RC4;
-  ssl_prefer_server_ciphers on;
-
   index  index.html;
 
-  access_log            /var/log/nginx/ssl-st2webui.access.log combined;
-  error_log             /var/log/nginx/ssl-st2webui.error.log;
+  access_log /proc/self/fd/1 combined;
+  error_log stderr;
 
-  add_header              Front-End-Https on;
-  add_header              X-Content-Type-Options nosniff;
+  resolver 127.0.0.11 valid=1s;
 
   location @apiError {
     add_header Content-Type application/json always;
@@ -51,7 +28,9 @@
 
     rewrite ^/api/(.*)  /$1 break;
 
-    proxy_pass            http://127.0.0.1:9101/;
+    set api_upstream ${ST2_API_URL};
+
+    proxy_pass            $api_upstream;
     proxy_read_timeout    90;
     proxy_connect_timeout 90;
     proxy_redirect        off;
@@ -81,7 +60,9 @@
 
     rewrite ^/stream/(.*)  /$1 break;
 
-    proxy_pass  http://127.0.0.1:9102/;
+    set stream_upstream ${ST2_STREAM_URL};
+
+    proxy_pass  $stream_upstream;
     proxy_set_header Host $host;
     proxy_set_header X-Real-IP $remote_addr;
     proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
@@ -109,7 +90,9 @@
 
     rewrite ^/auth/(.*)  /$1 break;
 
-    proxy_pass            http://127.0.0.1:9100/;
+    set auth_upstream ${ST2_AUTH_URL};
+
+    proxy_pass            $auth_upstream;
     proxy_read_timeout    90;
     proxy_connect_timeout 90;
     proxy_redirect        off;
@@ -134,4 +117,5 @@
     tcp_nopush on;
     tcp_nodelay on;
   }
+
 }
